tools/build-tool/code/tooldebian.pas

```diff
@@ -0,0 +1,192 @@
+unit ToolDebian;
+
+{$I castleconf.inc}
+
+interface
+
+uses
+  CastleFindFiles,
+  ToolArchitectures, ToolManifest, ToolPackage;
+
+type
+  { Package a project to a directory. }
+  TPackageDebian = class(TPackageDirectoryAbstract)
+  strict private
+    { Total size of all binaries in the project, required for Debian metadata }
+    BinariesSize: Int64;
+    { Helper to find total size of all files inside the folder,
+      To satisfy FindFile syntax has to be of Object }
+    procedure FoundFile(const FileInfo: TFileInfo; var StopSearch: Boolean);
+  public
+    procedure Make(const OutputProjectPath: String; const PackageFileName: String;
+      const Cpu: TCpu; const Manifest: TCastleManifest);
+  end;
+
+implementation
+uses
+  SysUtils, Process, {$ifdef UNIX} BaseUnix, {$endif}
+  CastleUtils, CastleFilesUtils, CastleDownload, CastleImages,
+  ToolCommonUtils;
+
+procedure TPackageDebian.FoundFile(const FileInfo: TFileInfo; var StopSearch: Boolean);
+begin
+  BinariesSize += FileInfo.Size;
+end;
+
+procedure TPackageDebian.Make(const OutputProjectPath: String; const PackageFileName: String;
+      const Cpu: TCpu; const Manifest: TCastleManifest);
+
+  function CpuToArchitectureString(const Cpu: TCpu): String;
+  begin
+    case Cpu of
+      { Architectures supported by https://packages.debian.org/buster/bash }
+      x86_64: Result := 'amd64';
+      aarch64: Result := 'arm64';
+      arm: begin
+          Result := 'armel';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "armel" and "armhf" in Debian.');
+        end;
+      i386: Result := 'i386';
+      mips: Result := 'mips';
+      mipsel: Result := 'mipsel';
+      powerpc64: begin
+          Result := 'ppc64el';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "ppc64el" and "ppc64" in Debian.');
+        end;
+      // "mips64el"
+      // "s390x"
+      { Unofficial Debian ports https://packages.debian.org/sid/bash }
+      m68k: Result := 'm68k';
+      riscv64: Result := 'riscv64';
+      sparc64: Result := 'sparc64';
+      // "alpha"
+      // "sh4"
+      // "x32" = X32 is an ABI for amd64/x86_64 CPUs using 32-bit integers, longs and pointers
+      else {powerpc,sparc,avr,jvm,i8086,riscv32,armeb}
+      begin
+        Result := CpuToString(Cpu);
+        WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is not officially supported by Debian.');
+      end;
+    end;
+  end;
+
+var
+  PathToExecutableLocal, PathToExecutableUnix: String;
+  PathToIconFileLocal, PathToIconFileUnix: String;
+  ShareDir: String;
+  TextWriter: TTextWriter;
+  OutString: String;
+  PackageFolder: String;
+begin
+  PackageFolder := 'castle-engine-output' + PathDelim + PackageFileName;
```

Comment:
Do not depend that `castle-engine-output` is a subdirectory of current working directory, do not even depend that it is a subdirectory of project directory. CGE build tool may be called from other directories (like any project subdirectory) and user can use `--output` to write the `castle-engine-output` anywhere.

So use `TempOutputPath(Project.Path)` above, like 

```
PackageFolder := TempOutputPath(Project.Path) + PackageFileName;
```

Reply:
Moreover I generally don't use term `Folder` in CGE code. Call it `PackageDir`. In CGE code I generally call things `XxxDir` (if they don't end with path delim) or `XxxPath` (if they do end with path delim).

Reply:
Fixed in 89fb8d670ddeef0fa8863b33811140d9084d2f8c